### PR TITLE
Export Database

### PR DESCRIPTION
**Pull request by root-beer**
ref: https://skyarch.backlog.jp/view/NC-650

### CODE REVIEW

File: lib/tasks/db.rake [hunk 1]
```diff
@@ -0,0 +1,35 @@
+require 'active_record'
+
+namespace :db do
+  namespace :data do
+    error_msg = 'Please specify the path. (e.g. rake db:data:load[./db/hoge.sql]'
+
+    desc "Dump the database to db/dbname.sql (path is optional)"
+    task :dump, [:path] => [:environment, :load_config] do |t, args|
```
Comment (skyarch-networks) [inline lib/tasks/db.rake]: __[rubocop]__
* Lint/UnusedBlockArgument: Unused block argument - `t`. If it's necessary, use `_` or `_t` as an argument name to indicate that it won't be used. [:ref](https://github.com/bbatsov/ruby-style-guide#underscore-unused-vars)

File: lib/tasks/db.rake [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      set_config
+
+      path = if args[:path]
+        args[:path]
```
Comment (skyarch-networks) [inline lib/tasks/db.rake]: __[rubocop]__
* Style/IndentationWidth: Use 2 (not -5) spaces for indentation. [:ref](https://github.com/bbatsov/ruby-style-guide#spaces-indentation)

File: lib/tasks/db.rake [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      else
+        "db/#{@config['database']}.sql"
+      end
+
+      cmd = "mysqldump -u#{@config['username']} -p#{@config['password']} #{@config['database']} --compatible=ansi > '#{path}'"
+      system(cmd)
+    end
+
+    desc "Load a .sql file into the database"
+    task :load, [:path] => [:environment, :load_config] do |t, args|
```
Comment (skyarch-networks) [inline lib/tasks/db.rake]: __[rubocop]__
* Lint/UnusedBlockArgument: Unused block argument - `t`. If it's necessary, use `_` or `_t` as an argument name to indicate that it won't be used. [:ref](https://github.com/bbatsov/ruby-style-guide#underscore-unused-vars)

**Comment by skyarch-networks** [comment]
Click here if you want to create rubocop's auto-correct Pull Request 
[Auto Correct by SideCI](https://sideci.com/github_repositories/31754199/issues/pull_request/322/diff/ab7b33b75b3d42eac952f2b9e660f0726475e083?open_modal=1#rubocop)

See the documentation if you want to get more information about auto-correct.
[Document](https://sideci.com/ja/docs/autofix-rubocop)

**Comment by root-beer** [comment]
Please review and merge PR. @Joeper214 🙇 

It looks like the error on SideCI, but there is no error in practice.

**Comment by Joeper214** [comment]
Okay I'll review it now.

**Comment by Joeper214** [comment]
Is it okay that we can add a short description about the database dump? and include the filename to be displayed in the UI?

**Comment by root-beer** [comment]
Okay. I'll add it.

**Comment by Joeper214** [comment]
よろしくお願いします。:bow:

File: app/controllers/app_settings_controller.rb [hunk 4]
```diff
@@ -128,6 +128,14 @@ def chef_keys
     @zipfile.close
   end
 
+  # GET /app_settings/export_db
+  def export_db
+    dbname = ActiveRecord::Base.configurations[Rails.env]['database']
+    path = Rails.root.join("db/#{dbname}.sql")
```
Comment (Joeper214) [inline app/controllers/app_settings_controller.rb]: Can we change the path to tmp/? or I think it's much better if we delete the sql file after dump. What do you think? @root-beer

**Comment by Joeper214** [comment]
Is it okay if it looks like this?

screen shot 2016-08-24 at 2 53 45 pm

**Comment by root-beer** [comment]
Looks good. 😄

**Comment by Joeper214** [comment]
Related to [this](https://github.com/skyarch-networks/skyhopper/pull/322#discussion_r75985924)

screen shot 2016-08-24 at 5 34 23 pm

File: app/controllers/app_settings_controller.rb (hunk 4 above)
Comment (root-beer) [inline app/controllers/app_settings_controller.rb]: I think so, too. I'll change it now.

**Comment by root-beer** [comment]
Is there any problem? @Joeper214

**Comment by Joeper214** [comment]
None so far. I'll merge this PR. Thanks!. @root-beer

**Comment by root-beer** [comment]
Thanks! 🙇

**Pull request merged**